core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -60,7 +60,7 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info).name = info.name }
+    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info)}
```

Comment:
It's not clear this is valid. You are updating the whole info object, not just the name, in the case that the ID existed. Why is this needed?